tm2/pkg/bft/consensus/reactor.go

```diff
@@ -242,9 +242,9 @@ func (conR *ConsensusReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte)
 			ps.ApplyHasVoteMessage(msg)
 		case *VoteSetMaj23Message:
 			cs := conR.conS
-			cs.mtx.Lock()
+			cs.mtx.RLock()
```

Comment:
OK this is an improvement, but would not have helped ultimately because the consensus machine was stuck on tryAddVote after receiving a vote, and the machine locks itself during messages. And blocking on event writes from the consensus engine *is* the behavior we want. (if we want any non-blocking or buffering then it should be so wrapped before plugging into the bft state machine). 

Anyways RLock() makes sense here so please continue with making such improvements where appropriate.

----

Go RWLock: a blocked Lock call excludes new readers from acquiring the lock. Where we use RLock() and starvation matters, I suggest we use `mtx.RLock() // must not starve` and add the comment, so nobody comes around and changes the mutex with a starving one (e.g. for many web clients reading, starving of writes). Here within the context of the consensus reactor I don't think it matters, because it is the ONE reader/writer. 

But other places like cs.GetRoundState(), which is potentially used by many web clients, should not starve the consensus engine, so would use `mtx.RLock() // must not starve` there for example.